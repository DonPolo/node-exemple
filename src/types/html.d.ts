declare module '*.html' {
  const value: Function;
  export default value;
}
