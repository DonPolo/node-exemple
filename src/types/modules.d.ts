declare module 'nexmo';

declare module 'chai-jest-diff';

declare module 'fake-fetch';
