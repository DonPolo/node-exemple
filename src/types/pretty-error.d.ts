/** Declaration file generated by dts-gen */
/*
export = pretty_error;

declare class pretty_error {
    constructor();

    alias(stringOrRx: any, alias: any): any;

    appendStyle(toAppend: any): any;

    config(c: any): any;

    filter(...args: any[]): any;

    filterParsedError(...args: any[]): any;

    getObject(e: any): any;

    removeAlias(stringOrRx: any): any;

    removeAllAliases(): any;

    removeAllFilters(): any;

    removeAllParsedErrorFilters(): any;

    removeFilter(...args: any[]): any;

    removeParsedErrorFilter(...args: any[]): any;

    render(e: any, logIt: any, useColors: any): any;

    setMaxItems(maxItems: any): any;

    skip(...args: any[]): any;

    skipNodeFiles(): any;

    skipPackage(...args: any[]): any;

    skipPath(...args: any[]): any;

    unskip(...args: any[]): any;

    unskipAll(): any;

    unskipAllPackages(): any;

    unskipAllPaths(): any;

    unskipNodeFiles(): any;

    unskipPackage(...args: any[]): any;

    unskipPath(...args: any[]): any;

    withColors(): any;

    withoutColors(): any;

    static start(): any;

    static stop(): any;

}

declare namespace pretty_error {
    namespace prototype {
        const renderer: any;

        const style: any;

        function alias(stringOrRx: any, alias: any): any;

        function appendStyle(toAppend: any): any;

        function config(c: any): any;

        function filter(...args: any[]): any;

        function filterParsedError(...args: any[]): any;

        function getObject(e: any): any;

        function removeAlias(stringOrRx: any): any;

        function removeAllAliases(): any;

        function removeAllFilters(): any;

        function removeAllParsedErrorFilters(): any;

        function removeFilter(...args: any[]): any;

        function removeParsedErrorFilter(...args: any[]): any;

        function render(e: any, logIt: any, useColors: any): any;

        function setMaxItems(maxItems: any): any;

        function skip(...args: any[]): any;

        function skipNodeFiles(): any;

        function skipPackage(...args: any[]): any;

        function skipPath(...args: any[]): any;

        function start(...args: any[]): any;

        function stop(): any;

        function unskip(...args: any[]): any;

        function unskipAll(): any;

        function unskipAllPackages(): any;

        function unskipAllPaths(): any;

        function unskipNodeFiles(): any;

        function unskipPackage(...args: any[]): any;

        function unskipPath(...args: any[]): any;

        function withColors(): any;

        function withoutColors(): any;

        namespace alias {
            const prototype: {
            };

        }

        namespace appendStyle {
            const prototype: {
            };

        }

        namespace config {
            const prototype: {
            };

        }

        namespace filter {
            const prototype: {
            };

        }

        namespace filterParsedError {
            const prototype: {
            };

        }

        namespace getObject {
            const prototype: {
            };

        }

        namespace removeAlias {
            const prototype: {
            };

        }

        namespace removeAllAliases {
            const prototype: {
            };

        }

        namespace removeAllFilters {
            const prototype: {
            };

        }

        namespace removeAllParsedErrorFilters {
            const prototype: {
            };

        }

        namespace removeFilter {
            const prototype: {
            };

        }

        namespace removeParsedErrorFilter {
            const prototype: {
            };

        }

        namespace render {
            const prototype: {
            };

        }

        namespace setMaxItems {
            const prototype: {
            };

        }

        namespace skip {
            const prototype: {
            };

        }

        namespace skipNodeFiles {
            const prototype: {
            };

        }

        namespace skipPackage {
            const prototype: {
            };

        }

        namespace skipPath {
            const prototype: {
            };

        }

        namespace start {
            const prototype: {
            };

        }

        namespace stop {
            const prototype: {
            };

        }

        namespace unskip {
            const prototype: {
            };

        }

        namespace unskipAll {
            const prototype: {
            };

        }

        namespace unskipAllPackages {
            const prototype: {
            };

        }

        namespace unskipAllPaths {
            const prototype: {
            };

        }

        namespace unskipNodeFiles {
            const prototype: {
            };

        }

        namespace unskipPackage {
            const prototype: {
            };

        }

        namespace unskipPath {
            const prototype: {
            };

        }

        namespace withColors {
            const prototype: {
            };

        }

        namespace withoutColors {
            const prototype: {
            };

        }

    }

}

*/
