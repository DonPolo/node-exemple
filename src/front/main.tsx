/* ################# */
/* #### Imports #### */
/* ################# */

import './sass/style.sass';

import ReactDOM from 'react-dom';
import React from 'react';

import App from './utils/app.util';

window.addEventListener('load', () => {
  const element = <App />;
  ReactDOM.render(element, document.getElementById('app'));
});
