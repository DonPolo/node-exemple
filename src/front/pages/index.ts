// Import the pages
import AddresponsePage from './addresponse.page';
import AnalyticsPage from './analytics.page';
import ChatPage from './chat.page';
import FilePage from './file.page';
import HomePage from './home.page';
import LoginPage from './login.page';
import NotfoundPage from './notfound.page';

export default {
  Addresponse: AddresponsePage,
  Analytics: AnalyticsPage,
  Chat: ChatPage,
  File: FilePage,
  Home: HomePage,
  Login: LoginPage,
  Notfound: NotfoundPage,
};
