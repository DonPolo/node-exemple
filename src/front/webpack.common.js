const CleanWebpackPlugin = require('clean-webpack-plugin');
const HtmlWebpackPlugin = require('html-webpack-plugin');
const path = require('path');

const config = {
	entry: {
		main: __dirname + '/JS/main.js'
	},
	output: {
		filename: '[name].js',
		path: path.join(__dirname, '../../assets/js')
	},
	module: {
		rules: [
			{
				test: /\.css$/,
				use: ['style-loader', 'css-loader']
			},
			{
				test: /\.js$/,
				use: 'babel-loader'
      },
      {
        test: /\.sass$/,
        use: ['style-loader', 'css-loader', 'sass-loader']
      }
		]
	},
	plugins: [
    new CleanWebpackPlugin(),
    new HtmlWebpackPlugin({
      title: 'Production'
    })
  ]
};

module.exports = config;
