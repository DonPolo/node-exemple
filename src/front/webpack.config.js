const CleanWebpackPlugin = require('clean-webpack-plugin');
const path = require('path');

const config = {
  entry: {
    main: __dirname + '/main.tsx',
  },
  output: {
    filename: '[name].js',
    path: path.join(__dirname, '../../assets/js'),
  },
  resolve: {
    extensions: ['.ts', '.tsx', '.js', '.json'],
  },
  module: {
    rules: [
      {
        test: /\.css$/,
        use: ['style-loader', 'css-loader'],
      },
      {
        test: /\.tsx$/,
        exclude: /node_modules/,
        use: 'awesome-typescript-loader',
      },
      {
        test: /\.ts$/,
        exclude: /node_modules/,
        use: 'ts-loader',
      },
      {
        test: /\.sass$/,
        use: ['style-loader', 'css-loader', 'sass-loader'],
      },
    ],
  },
  node: {
    fs: 'empty',
  },
  plugins: [new CleanWebpackPlugin()],
};

module.exports = config;
