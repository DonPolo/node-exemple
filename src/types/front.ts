import {
  ParsedResponseBtn,
  ParsedResponseDropdown,
  ParsedResponseLink,
  ParsedResponseMedia,
  ParsedResponseText,
} from './types.util';

export interface User {
  pseudo: string;
  pass: string;
}

export interface FileInfos {
  intent: string;
  cat: 'training' | 'response';
}

export interface ChatResponse {
  res:
    | ParsedResponseBtn
    | ParsedResponseDropdown
    | ParsedResponseLink
    | ParsedResponseMedia
    | ParsedResponseText;
  pos: string;
  id: number;
}
