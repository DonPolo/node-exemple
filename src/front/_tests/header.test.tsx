import React from 'react';
import renderer from 'react-test-renderer';
import Header from '../components/header.component';
import { User } from '../../types/front';
import { BrowserRouter } from 'react-router-dom';

const user: User = {
  pseudo: 'Admin',
  pass: '',
};

// Unusefull
describe('Header style testing', () => {
  test('No regression', () => {
    const header = renderer.create(
      <BrowserRouter>
        <Header user={user} nav={2} />
      </BrowserRouter>,
    );

    const tree = header.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
