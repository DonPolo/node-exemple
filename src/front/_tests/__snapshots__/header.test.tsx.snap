// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header style testing No regression 1`] = `
<header>
  <div>
    <img
      src="/pic/lifee.png"
    />
    <h1>
      Lifee Admin
    </h1>
  </div>
  <ul>
    <a
      className=""
      href="/chat"
      onClick={[Function]}
    >
      <li>
        Chat
      </li>
    </a>
    <a
      className="selected"
      href="/"
      onClick={[Function]}
    >
      <li>
        Training & Responses
      </li>
    </a>
    <a
      className=""
      href="/addresponse"
      onClick={[Function]}
    >
      <li>
        Add response
      </li>
    </a>
    <a
      className=""
      href="/analytics"
      onClick={[Function]}
    >
      <li>
        Analytics
      </li>
    </a>
    <a
      className=""
      href="/webapp/disconnect"
    >
      <li>
        Disconnect
      </li>
    </a>
  </ul>
  <span>
    <h2>
      Admin
    </h2>
  </span>
</header>
`;
